app/web_framework/cluster_master_application.py

```diff
@@ -417,7 +420,7 @@ class _SlavesShutdownHandler(_ClusterMasterBaseAPIHandler):
     @authenticated
     def post(self):
         shutdown_all = self.decoded_body.get('shutdown_all')
-        slaves_to_shutdown = self._cluster_master.all_slaves_by_id().keys() if shutdown_all else\
+        slaves_to_shutdown = SlaveRegistry.singleton().get_all_slaves_by_id().keys() if shutdown_all else\
```

Comment:
Use `( )` instead of `\`. But in this case I would split into a regular `if / else` clause.